include/snowflake/basic_types.h

```diff
@@ -9,15 +9,18 @@
 extern "C" {
 #endif
 
+#include <limits.h>
+#include <float.h>
+#include <math.h>
 #include "platform.h"
 
 /**
  * Supported data types
  */
 typedef char int8;
 typedef unsigned char uint8;
-typedef unsigned long int uint32;
-typedef long int int32;
+typedef unsigned int uint32;
+typedef int int32;
```

Comment:
In the future we'll probably want to check type sizes on a particular platform to ensure that our types are the correct size.